File: app/src/main/java/eu/darken/bluemusic/main/ui/config/LaunchAppAdapter.java
package eu.darken.bluemusic.main.ui.config;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import eu.darken.bluemusic.R;
import eu.darken.bluemusic.util.AppTool;

public class LaunchAppAdapter extends BaseAdapter {
    private final Context context;
    private final List<AppTool.Item> apps;

    public LaunchAppAdapter(Context context, List<AppTool.Item> apps) {
        this.context = context;
        this.apps = apps;
    }

    @Override
    public int getCount() {
        return apps.size();
    }

    @Override
    public AppTool.Item getItem(int i) {
        return apps.get(i);
    }

    @Override
    public long getItemId(int i) {
        return apps.get(i).hashCode();
    }

    @SuppressLint("InflateParams")
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(parent.getContext()).inflate(R.layout.adapter_app_line, null);
            holder = new ViewHolder();
            ButterKnife.bind(holder, convertView);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        holder.bind(getItem(position));
        return convertView;
    }


    static class ViewHolder {
        @BindView(R.id.icon) ImageView icon;
        @BindView(R.id.name) TextView appName;
        @BindView(R.id.pkg) TextView packageName;

        public void bind(AppTool.Item item) {
            appName.setText(item.getAppName());
            packageName.setText(item.getPackageName());
            icon.setImageDrawable(item.getAppIcon());
        }
    }
}

File: app/src/main/java/eu/darken/bluemusic/main/ui/config/LaunchAppAdapter.kt
package eu.darken.bluemusic.main.ui.config

//class LaunchAppAdapter(private val apps: List<AppTool.Item>) : BaseAdapter() {
//    override fun getCount(): Int = apps.size
//
//    override fun getItem(i: Int): AppTool.Item = apps[i]
//
//    override fun getItemId(i: Int): Long = apps[i].hashCode().toLong()
//
//    @SuppressLint("InflateParams") override fun getView(position: Int, _convertView: View?, parent: ViewGroup): View {
//        var convertView = _convertView
//        val holder: ViewHolder
//        if (convertView != null) {
//            holder = convertView.tag as ViewHolder
//        } else {
//            convertView = LayoutInflater.from(parent.context).inflate(R.layout.adapter_app_line, null)
//            holder = ViewHolder(convertView)
//            convertView.tag = holder
//        }
//        holder.bind(getItem(position))
//        return convertView!!
//    }
//
//    class ViewHolder(layout: View) {
//        val icon = layout.findViewById<ImageView>(R.id.icon)
//        val appName = layout.findViewById<TextView>(R.id.name)
//        val packageName = layout.findViewById<TextView>(R.id.pkg)
//
//        fun bind(item: AppTool.Item) {
//            appName.text = item.appName
//            packageName.text = item.pkgName
//            icon.setImageDrawable(item.appIcon)
//        }
//    }
//}